pkg/apis/pipeline/v1alpha1/taskrun_types.go

```diff
@@ -43,7 +43,19 @@ type TaskRunSpec struct {
 	// +optional
 	Generation int64 `json:"generation,omitempty"`
 	// +optional
-	ServiceAccount string `json:"serviceAccount"`
+	ServiceAccount string `json:"serviceAccount,omitempty"`
+	// +optional
+	PVCName string `json:"pvcName,omitempty"`
+	// +optional
+	PreBuiltSteps []TaskBuildStep `json:"preBuildSteps,omitempty"`
```

Comment:
Do we have to have these steps in the definition of the taskRun type as a field, can we not append them in the appropriate order while creating the taskRun

Reply:
> If not, can we somehow make it obvious we don't want users to provide them?

How about add validation in taskrun to check if owner reference is `pipelinerun Kind` then post build and pre build steps should be allowed else throw error.

Reply:
If we don't allow user to provide those field, it shouldn't be in the spec (should it ?), but I guess they are required for pipeline/pipelinerun to populate them… so… yeah most likely a validation error could work there.

Is there a use case for a user to provides those in addition to *normal* steps ?

Reply:
> Is there a use case for a user to provides those in addition to normal steps ?

@vdemeester: @shashwathi and i discussed, and we were thinking that maybe a use case where a user wants to debug tasks that are part of a pipeline in isolation? it might be a bit of a stretch tho!

i think adding `paths` to the interface instead of the `builtSteps` makes it a bit clearer hopefully!